src/utils.test.js

```diff
@@ -60,4 +62,59 @@ describe('utils', () => {
       expect(dataHubParams.hash).toBe('#hash1#hash2');
     });
   });
+
+  describe('setInterval custom hook', () => {
+    it('should set an interval and update an outside value', () => {
+      jest.useFakeTimers();
+      let counter = 0;
+      renderHook(() =>
+        useInterval({
+          runOnStart: false,
+          delay: 100,
+          callback: () => {
+            counter++;
+          },
+        }),
+      );
+
+      jest.runOnlyPendingTimers();
+      expect(counter).toBe(1);
+    });
+
+    it('should set an interval and pause when delay is null', () => {
+      jest.useFakeTimers();
+      let counter = 0;
+      const { rerender } = renderHook(
+        ({ delay }) =>
+          useInterval({
+            runOnStart: false,
+            delay: delay,
+            callback: () => {
+              counter++;
+            },
+          }),
+        { initialProps: { delay: 100 } },
+      );
+      jest.advanceTimersByTime(600);
+      rerender({ delay: null });
+      jest.advanceTimersByTime(800);
+      expect(counter).toBe(6);
+    });
+
+    it('should set an interval and run callback before start', () => {
+      jest.useFakeTimers();
+      let counter = 0;
+      renderHook(() =>
+        useInterval({
+          runOnStart: true,
+          delay: 100,
+          callback: () => {
+            counter++;
+          },
+        }),
+      );
+      jest.advanceTimersByTime(200);
```

Comment:
What do you think about also verify before moving in time?

```suggestion
      expect(counter).toBe(1);
      jest.advanceTimersByTime(200);
```

Reply:
Fixed.